src/java.base/share/classes/sun/security/ssl/CertificateAuthoritiesExtension.java

```diff
@@ -122,11 +123,17 @@ private static List<byte[]> getEncodedAuthorities(
             return authorities;
         }
 
-        X500Principal[] getAuthorities() {
+        X500Principal[] getAuthorities() throws SSLException{
             X500Principal[] principals = new X500Principal[authorities.size()];
-            int i = 0;
-            for (byte[] encoded : authorities) {
-                principals[i++] = new X500Principal(encoded);
+
+            try {
+                int i = 0;
+                for (byte[] encoded : authorities) {
+                    principals[i++] = new X500Principal(encoded);
+                }
+            } catch (IllegalArgumentException iae) {
+                throw new SSLException("X500Principal could not be parsed " +
+                        "successfully", iae);
```

Comment:
Is it ok to throw a general `SSLException` here? Or do you need to call `TransportContext.fatal()` so that additional cleanup happens? Perhaps @XueleiFan would know.

Reply:
It is not easy to understand the final behavior if throwing SSLException here.  I would like to call `TransportContext.fatal()` directly to make the behavior more accuracy, by using Alert.DECODE_ERROR.

Reply:
You will need to pass in `TransportContext` as a parameter if you do that, unless you go back changing the callers of `getAuthorities()` to catch `IllegalArgumentException`. I'm now thinking it is better for the callers of `getAuthorities()` to catch `IllegalArgumentException` and then call `fatal`.

One other minor comment:

- I would remove the word "successfully" as this is a failure case so it is implied.